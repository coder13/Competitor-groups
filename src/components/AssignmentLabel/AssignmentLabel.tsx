import { AssignmentCode } from '@wca/helpers';
import Assignments from '@/lib/assignments';

const Container = ({ children, className }: React.PropsWithChildren<{ className?: string }>) => (
  <span className={`${className} px-[6px] py-[4px] rounded-md bg-blue-100`}>{children}</span>
);

interface AssignmentLabelProps {
  assignmentCode: AssignmentCode;
}

export function AssignmentLabel({ assignmentCode }: AssignmentLabelProps) {
  if (assignmentCode.match(/judge/i)) {
    return (
      <Container className="bg-blue-200">
        {assignmentCode === 'staff-judge' ? 'Judge' : assignmentCode.replace('staff-', '')}
      </Container>
    );
  }

  const config = Assignments.find((a) => a.id === assignmentCode);

  switch (assignmentCode) {
    case 'competitor':
      return <Container className="bg-green-200">{config?.name}</Container>;
    case 'staff-scrambler':
      return <Container className="bg-yellow-200">{config?.name}</Container>;
    case 'staff-runner':
      return <Container className="bg-orange-200">{config?.name}</Container>;
    case 'staff-dataentry':
      return <Container className="bg-cyan-200">{config?.name}</Container>;
    case 'staff-announcer':
      return <Container className="bg-violet-200">{config?.name}</Container>;
    case 'staff-delegate':
      return <Container className="bg-purple-200">{config?.name}</Container>;
    case 'staff-stagelead':
      return <Container className="bg-purple-800">{config?.name}</Container>;
    default:
      return <Container>{config ? config.name : assignmentCode.replace('staff-', '')}</Container>;
  }
}
