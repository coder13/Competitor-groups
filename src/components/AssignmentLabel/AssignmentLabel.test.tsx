import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { AssignmentLabel } from './AssignmentLabel';

describe('components/AssignmentLabel', () => {
  it('should render correctly when assignmentCode is "competitor"', () => {
    const { baseElement } = render(<AssignmentLabel assignmentCode="competitor" />);
    expect(baseElement).toMatchInlineSnapshot(`
      <body>
        <div>
          <span
            class="px-[6px] py-[4px] rounded-md bg-green-200"
          >
            Competitor
          </span>
        </div>
      </body>
    `);
  });

  it('should render correctly when assignmentCode is "staff-judge"', () => {
    const { baseElement } = render(<AssignmentLabel assignmentCode="staff-judge" />);
    expect(baseElement).toMatchInlineSnapshot(`
      <body>
        <div>
          <span
            class="px-[6px] py-[4px] rounded-md bg-blue-200"
          >
            Judge
          </span>
        </div>
      </body>
    `);
  });

  it('should render correctly when assignmentCode is "staff-runner"', () => {
    const { baseElement } = render(<AssignmentLabel assignmentCode="staff-runner" />);
    expect(baseElement).toMatchInlineSnapshot(`
      <body>
        <div>
          <span
            class="px-[6px] py-[4px] rounded-md bg-orange-200"
          >
            Runner
          </span>
        </div>
      </body>
    `);
  });

  it('should render correctly when assignmentCode is "staff-scrambler"', () => {
    const { baseElement } = render(<AssignmentLabel assignmentCode="staff-scrambler" />);
    expect(baseElement).toMatchInlineSnapshot(`
      <body>
        <div>
          <span
            class="px-[6px] py-[4px] rounded-md bg-yellow-200"
          >
            Scrambler
          </span>
        </div>
      </body>
    `);
  });

  it('should render correctly when assignmentCode is non-standard', () => {
    const { baseElement } = render(<AssignmentLabel assignmentCode="staff-dataentry" />);
    expect(baseElement).toMatchInlineSnapshot(`
      <body>
        <div>
          <span
            class="px-[6px] py-[4px] rounded-md bg-blue-100"
          >
            staff-dataentry
          </span>
        </div>
      </body>
    `);
  });
});
