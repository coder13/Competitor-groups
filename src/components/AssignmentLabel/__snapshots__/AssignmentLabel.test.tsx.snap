// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AssignmentLabel should render correctly when assignmentCode is "competitor" 1`] = `
<body>
  <div>
    <span
      class="bg-green-200 px-[6px] py-[4px] rounded-md bg-blue-100"
    >
      competitor
    </span>
  </div>
</body>
`;

exports[`components/AssignmentLabel should render correctly when assignmentCode is "staff-announcer" 1`] = `
<body>
  <div>
    <span
      class="bg-violet-200 px-[6px] py-[4px] rounded-md bg-blue-100"
    >
      announcer
    </span>
  </div>
</body>
`;

exports[`components/AssignmentLabel should render correctly when assignmentCode is "staff-dataentry" 1`] = `
<body>
  <div>
    <span
      class="bg-cyan-200 px-[6px] py-[4px] rounded-md bg-blue-100"
    >
      dataentry
    </span>
  </div>
</body>
`;

exports[`components/AssignmentLabel should render correctly when assignmentCode is "staff-delegate" 1`] = `
<body>
  <div>
    <span
      class="bg-purple-200 px-[6px] py-[4px] rounded-md bg-blue-100"
    >
      delegate
    </span>
  </div>
</body>
`;

exports[`components/AssignmentLabel should render correctly when assignmentCode is "staff-judge" 1`] = `
<body>
  <div>
    <span
      class="bg-blue-200 px-[6px] py-[4px] rounded-md bg-blue-100"
    >
      judge
    </span>
  </div>
</body>
`;

exports[`components/AssignmentLabel should render correctly when assignmentCode is "staff-other" 1`] = `
<body>
  <div>
    <span
      class="undefined px-[6px] py-[4px] rounded-md bg-blue-100"
    >
      other
    </span>
  </div>
</body>
`;

exports[`components/AssignmentLabel should render correctly when assignmentCode is "staff-runner" 1`] = `
<body>
  <div>
    <span
      class="bg-orange-200 px-[6px] py-[4px] rounded-md bg-blue-100"
    >
      runner
    </span>
  </div>
</body>
`;

exports[`components/AssignmentLabel should render correctly when assignmentCode is "staff-scrambler" 1`] = `
<body>
  <div>
    <span
      class="bg-yellow-200 px-[6px] py-[4px] rounded-md bg-blue-100"
    >
      scrambler
    </span>
  </div>
</body>
`;

exports[`components/AssignmentLabel should render correctly when assignmentCode is "staff-stagelead" 1`] = `
<body>
  <div>
    <span
      class="bg-purple-800 px-[6px] py-[4px] rounded-md bg-blue-100"
    >
      stagelead
    </span>
  </div>
</body>
`;
